File: Client/src/com/badlogic/core/factory/bonuses/BonusFactory.java
package com.badlogic.core.factory.bonuses;

import com.badlogic.core.factory.AbstractFactory;
import com.badlogic.game.GameManager;

import java.awt.image.BufferedImage;

public class BonusFactory implements AbstractFactory<Bonus> {
    @Override
    public Bonus create(String bonusType, GameManager gameManager, BufferedImage sprite) {
        switch (bonusType) {
            case BonusType.AMMO:
                return new AmmoBonus(gameManager, sprite);
            case BonusType.HEALTH:
                return new HealthBonus(gameManager, sprite);
            case BonusType.SPEED:
                return new SpeedBonus(gameManager, sprite);
            default:
                return new NullBonus(gameManager, sprite);
        }
    }
}

File: Server/Game/Bonuses/BonusFactory.cs
namespace Server.Game.Bonuses
{
    public static class BonusFactory
    {
        public static Bonus GetBonus(string bonusType, int bonusAmount = 0, int lifespan = 0)
        {
            switch (bonusType)
            {
                case BonusType.Health:
                    return new HealthBonus(bonusType, bonusAmount, lifespan);
                case BonusType.Ammo:
                    return new AmmoBonus(bonusType, bonusAmount, lifespan);
                case BonusType.Speed:
                    return new SpeedBonus(bonusType, bonusAmount, lifespan);
                default:
                    return null;
            }
        }
    }
}